geode-management/src/main/java/org/apache/geode/management/api/RestfulEndpoint.java

```diff
@@ -17,7 +17,23 @@
 
 import com.fasterxml.jackson.annotation.JsonIgnore;
 
-public interface RestfulEndpoint {
+import org.apache.geode.lang.Identifiable;
+
+/**
+ * established the minimum requirements for a restful service request -- must have a unique id, a
+ * scope, an endpoint to list all of its kind, an endpoint to list itself, and a type declaration
+ * for the return type when list or other operations are performed
+ */
+
+// "type parameter is never used" warning is suppressed, because actually it is used to create
+// the linkage between request and response type in the signature of ClusterManagementService.list
+@SuppressWarnings("unused")
+public interface RestfulEndpoint<R extends RuntimeResponse> extends Identifiable<String> {
+  /** should return null if no group */
+  String getGroup();
```

Comment:
RestEndPoint is just an interface to get the url, Why it needs to be mixed with group concept?

Reply:
The return type of the REST endpoint is just as important as the URL.

I did initially make separate interfaces for those two facets and it seems to create a lot of needless parallel structure, since everything we expose needs to have both a URL and a return type.

Reply:
It looks like this new RestfulEndpoint is the old "RestfulEndpoint" combined with "CacheElement". probably just some naming changes would make things more understandable.

Reply:
it's just this "getGroup" and "setGroup" is defined in multiple places: CacheElement, RuntimeResponse, and RestfulEndPoint which doesn't feel right. Would be nice to keep them in one place.

Reply:
reverted back to just CacheElement everywhere for this PR.